frontend/components/market/TechnicalAnalysis.tsx

```diff
@@ -0,0 +1,361 @@
+"use client";
+import React, { useState, useEffect } from 'react';
+import { motion } from 'framer-motion';
+import { TrendingUp, TrendingDown, BarChart3, Activity, Target, AlertTriangle } from 'lucide-react';
+
+interface TechnicalIndicator {
+  name: string;
+  value: string;
+  signal: 'buy' | 'sell' | 'neutral';
+  strength: number; // 0-100
+  description: string;
+}
+
+interface MarketData {
+  symbol: string;
+  price: string;
+  change24h: string;
+  volume: string;
+  marketCap: string;
+  indicators: TechnicalIndicator[];
+  support: string;
+  resistance: string;
+  rsi: number;
+  macd: { value: string; signal: 'bullish' | 'bearish' };
+}
+
+interface TechnicalAnalysisProps {
+  symbol?: string;
+  timeframe?: '1h' | '4h' | '1d' | '1w';
+  onTimeframeChange?: (timeframe: string) => void;
+}
+
+const TechnicalAnalysis: React.FC<TechnicalAnalysisProps> = ({
+  symbol = 'NEAR',
+  timeframe = '1d',
+  onTimeframeChange
+}) => {
+  const [marketData, setMarketData] = useState<MarketData | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [selectedIndicator, setSelectedIndicator] = useState<string | null>(null);
+
+  // Mock market data
+  const mockData: MarketData = {
```

Comment:
lets pull this market data from the agent itself so that its dynamic in the next PR.